### PR TITLE
ported iterkeys to py3 syntax

### PR DESCRIPTION
**Pull request by Shaps**
##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
vsphere_guest

##### ANSIBLE VERSION
N/A
##### SUMMARY
Ported `iterkeys()` to py3 syntax
Addresses ansible/ansible#18507

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @Shaps for this PR. This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed. Thanks for your patience.

Core team: please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with 'needs_revision' or merge as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/vmware/vsphere_guest.py [hunk 1]
```diff
@@ -986,7 +988,7 @@ def reconfigure_vm(vsphere_client, vm, module, esxi, resource_pool, cluster_name
         disk_num = 0
         dev_changes = []
         disks_changed = {}
-        for disk in sorted(vm_disk.iterkeys()):
+        for disk in sorted(iterkeys(vm_disk)):
```
Comment (abadger) [inline cloud/vmware/vsphere_guest.py]: We can just do this:
``` python

for disk in sorted(vm_disks):
```

There's no need to use iterkeys() from the six module.

**Comment by abadger** [comment]
needs_revision

**Comment by gregdek** [comment]
Thanks @Shaps for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by Shaps** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @Shaps for this PR. This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed. Thanks for your patience.

Core team: please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with 'needs_revision' or merge as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by abadger** [comment]
Thanks @Shaps, I've merged this into devel and stable-2.2.  It will either make it into 2.2.1 or 2.2.2

**Pull request merged**